bin/foreman-node

```diff
@@ -61,15 +65,23 @@ def plain_grains(minion)
   # We have to get the grains from the cache, because the client
   # is probably running 'state.highstate' right now.
 
-  result = IO.popen(['salt-run', '-l', 'quiet', '--output=json', 'cache.grains', minion]) do |io|
-    io.read
-  end
+  if SETTINGS[:filecache]
+    file = File.open("/var/cache/salt/master/minions/#{minion}/data.p", "rb")
```

Comment:
I guess, the path to the data.p should be part of the foreman.yaml settings file.

Reply:
Fixed in latest commit